src/node_osrm_support.hpp

```diff
@@ -420,14 +420,97 @@ inline bool parseCommonParameters(const v8::Local<v8::Object> &obj, ParamType &p
     if (obj->Has(Nan::New("annotations").ToLocalChecked()))
     {
         auto annotations = obj->Get(Nan::New("annotations").ToLocalChecked());
-        if (annotations->IsBoolean())
+        if (!annotations->IsArray())
```

Comment:
If annotations is not an array it should be a boolean. Allowing `annoations="distance"` increases the API surface too much IMHO.

What should be checked here is should be something like:
```
if (annotations->IsBoolean()) {
params->annotations = annotations->BooleanValue();
} else if (annotations->IsArrary()) {
params->annotations_type = .....
} else {
}
```